packages/docs/docs/guides/8-glossary.md

```diff
@@ -0,0 +1,108 @@
+
+# Request Protocol
+
+## Action
+
+An action is signed data added by a request's stakeholder into the Request Protocol that creates or updates the state of a request. The creation of a request is considered as an action and a request can be represented by a list of actions.
+
+## Balance
+
+When using a payment network, the balance is the current amount paid for a request. This balance is not the current balance of a specific account but is determined by the payment detection method of the payment network used.
+
+A request with no payment network provided doesn't have a balance.
+
+## Confirmed/Pending action
+
+The Protocol relies on other blockchain technologies to ensure data immutability. Most blockchain doesn't offer transaction instant finality. This means that when performing an action on the request, this action can't directly be confirmed as effective.
+
+As long as the action hasn't been persisted and is not confirmed, the action is marked as "pending".
+
+## Encryption provider
+
+An encryption provider is an object that allows to abstract the encryption of actions. It can be used to perform the action's encryption on a distant server.
+
+## Extension
+
+An extension is a set of actions that extends the feature of a request. A request without extension is a basic request for payment with a payee, a currency and a requested amount. The extension allows for more advanced features.
+
+## Identity
+
+The identity is what defines a stakeholder of a request that allows signing or encrypting the request actions. The identity is the public data that identifies the stakeholder.
+
+## Payment Detection
+
+A payment detection is a method defined by the payment network to determine the current balance of a request.
+
+## Payment Network
+
+A payment network is a predefined set of rules to agree on the balance of a request. The payment network is defined during the creation of the request.
+
+A payment network is generally related to one currency but it's not always the case (the Declarative payment network is currency agnostic)
+
+## Request Data
+
+The request data is the current state of a request, the data of the request after having applied all the confirmed actions on it.
+
+## Request Id
+
+The request Id is the number that uniquely identifies a request. This number is computed from the hash of the request creation action.
+
+## Signature Provider
+
+A signature provider is an object that allows to abstract identity management and action signatures. It can be used to perform the action's signature on a distant server.
+
+## Stakeholder
+
+A request stakeholder is a party involved with the request. Stakeholders are generally the payer and the payee of the request and perform actions on it but for encrypted requests, stakeholders are any party that can read the request.
```

Comment:
```suggestion
A request stakeholder is a party involved with the request. Stakeholders are generally the payer and the payee of the request and perform actions on it but for encrypted requests, stakeholders are any party that has an interest reading the request.
```

Reply:
otherwise all the user are stakeholder of all the clear requests

Reply:
I precised `for encrypted requests` but it's a good suggestion